test/server/connection_handler_test.cc

```diff
@@ -81,7 +81,7 @@ class ConnectionHandlerTest : public testing::Test, protected Logger::Loggable<L
     TestListener* listener =
         new TestListener(*this, tag, bind_to_port, hand_off_restored_destination_connections, name,
                          socket_type, listener_filters_timeout);
-    listener->moveIntoListBack(TestListenerPtr{listener}, listeners_);
```

Comment:
Also see here: Even I have the knowledge here it takes me a while to reason about.